packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -48,6 +55,202 @@ export async function fetchSnapshot(
+                    redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                    dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                    tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                    secureConntime = (indResTime.secureConnectionStart > 0) ?
+                        (indResTime.connectEnd - indResTime.secureConnectionStart) : 0;
+                    responseTime = indResTime.responseEnd - indResTime.responseStart;
+                    fetchStToRespEndTime = (indResTime.fetchStart > 0) ?
+                        (indResTime.responseEnd - indResTime.fetchStart) : 0;
+                    reqStToRespEndTime = (indResTime.requestStart > 0) ?
+                        (indResTime.responseEnd - indResTime.requestStart) : 0;
+                    networkTime = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : 0;
+                    if (spReqDuration) {
+                        networkTime = networkTime - parseInt(spReqDuration, 10);
+                    }
+                    break;
+                }
+            }
+
+            const { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize } = evalBlobsAndTrees(snapshot);
+            const clientTime = networkTime ? overallTime - networkTime : undefined;
+
+            // There are some scenarios in ODSP where we cannot cache, trees/latest will explicitly tell us when we
+            // cannot cache using an HTTP response header.
+            const canCache = response.headers.get("disablebrowsercachingofusercontent") !== "true";
+            // There maybe no snapshot - TreesLatest would return just ops.
+            const sequenceNumber: number = (snapshot.trees && (snapshot.trees[0] as any).sequenceNumber) ?? 0;
+            const seqNumberFromOps = snapshot.ops && snapshot.ops.length > 0 ?
+                snapshot.ops[0].sequenceNumber - 1 :
+                undefined;
+
+            const value: ISnapshotCacheValue = { snapshot, sequenceNumber };
+            if (!Number.isInteger(sequenceNumber)
+                || seqNumberFromOps !== undefined && seqNumberFromOps !== sequenceNumber) {
+                logger.sendErrorEvent({ eventName: "fetchSnapshotError", sequenceNumber, seqNumberFromOps });
+                value.sequenceNumber = undefined;
+            } else if (canCache) {
+                const fluidEpoch = response.headers.get("x-fluid-epoch");
+                assert(fluidEpoch !== undefined, "Epoch  should be present in response");
+                const valueWithEpoch: IVersionedValueWithEpoch = {
+                    value,
+                    fluidEpoch,
+                    version: 2,
+                };
+                putInCache(valueWithEpoch).catch((error) => {
+                    logger.sendErrorEvent({ eventName: "cachePutError", type: snapshotKey }, error);
```

Comment:
I think you can move responsibility of logging errors to the caller, and make it simpler. I believe the main path already goes through epochTracker.put that already logs errors. So really logging is only interesting for prefetching path and it might take more generic form of whole process fails, here is an error object to log

Reply:
It's already logged there indirectly through PerformanceEvent.timedExecAsync and the fact that we await cached promise, so I believe there is nothing to do really